aimmo-game/simulation/world_map.py

```diff
@@ -149,9 +157,25 @@ def get_random_spawn_location(self):
     def can_move_to(self, target_location):
         if not self.is_on_map(target_location):
             return False
+        cell = self.get_cell(target_location)
 
+        return cell.habitable and not cell.avatar and len(cell.actions) == 1
+
+    def attackable_avatar(self, target_location):
+        '''
+        Return the avatar attackable at the given location, or None.
+        '''
+        if not self.is_on_map(target_location):
+            return False
         cell = self.get_cell(target_location)
-        return cell.habitable and not cell.avatar
+
+        if cell.avatar:
+            return cell.avatar
+
+        if len(cell.actions) == 1:
+            return cell.actions[0].avatar
```

Comment:
This causes an avatar to attack itself if this is the only action on target_location.